gabbi/case.py

```diff
@@ -187,13 +190,27 @@ def _clean_query_value(self, value):
         return value
 
     def _environ_replace(self, message):
-        """Replace an indicator in a message with the environment value."""
+        """Replace an indicator in a message with the environment value.
+
+        If value can be a number, cast it as such. If value is a form of
+        "null", "true", or "false" cast it to None, True, False.
+        """
         value = re.sub(self._replacer_regex('ENVIRON'),
                        self._environ_replacer, message)
-        if value == "False":
+        try:
+            if '.' in value:
+                value = float(value)
+            else:
+                value = int(value)
+            return value
+        except ValueError:
+            pass
+        if value.lower() == "false":
             return False
-        if value == "True":
+        if value.lower() == "true":
             return True
+        if value.lower() == "null":
+            return None
```

Comment:
This seems like we're reinventing JSON/YAML parsing; do we really wanna do that?

PS: Would it be silly to ask for caching `value.lower()` in a local variable?